cache/Cache.js

```diff
@@ -0,0 +1,63 @@
+const EventEmitter = require('events')
+const _createRedisClient = require('./util/createRedisClient')
+const msgpack = require('msgpack-lite')
+
+// default entry expiration time in seconds
+const DEFAULT_EXPIRATION = 86400
+
+class Cache extends EventEmitter {
+  constructor ({ logger, nodes, defaultTtl, redisClientOptions }) {
+    super()
+    this._logger = logger
+    this._defaultTtl = defaultTtl || DEFAULT_EXPIRATION
+
+    this._onReady = () => this.emit('ready')
+    this._onError = (err) => this.emit('error', err)
+    this._onReconnecting = () => this.emit('reconnecting')
+    this._onNodeError = (err) => this.emit('node error', err)
```

Comment:
I'm personally not a big fan of events with spaces. Maybe `node-error`? What do you think?